packages/angular_devkit/schematics_cli/bin/schematics.ts

```diff
@@ -354,34 +338,48 @@ function getUsage(): string {
 
 /** Parse the command line. */
 const booleanArgs = [
-  'allowPrivate',
   'allow-private',
   'debug',
   'dry-run',
-  'dryRun',
   'force',
   'help',
   'list-schematics',
-  'listSchematics',
   'verbose',
   'interactive',
 ];
 
-function parseArgs(args: string[] | undefined): minimist.ParsedArgs {
-  return minimist(args, {
+/** Parse the command line. */
+function parseArgs(args: string[]): { _: string[] } & Record<string, unknown> {
+  const { $0, _, ...options } = yargsParser(args, {
     boolean: booleanArgs,
-    alias: {
-      'dryRun': 'dry-run',
-      'listSchematics': 'list-schematics',
-      'allowPrivate': 'allow-private',
-    },
     default: {
       'interactive': true,
       'debug': null,
-      'dryRun': null,
+      'dry-run': null,
+    },
+    configuration: {
+      'unknown-options-as-args': false,
+      'dot-notation': false,
+      'boolean-negation': true,
+      'strip-aliased': true,
+      'camel-case-expansion': false,
     },
-    '--': true,
   });
+
+  // Camelize options as yargs will return the object in kebab-case when camel casing is disabled.
+  const camelCasedOptions: Record<string, unknown> = {};
+  for (const [key, value] of Object.entries(options)) {
+    if (/A-Z/.test(key)) {
+      throw new Error(`Unknown argument ${key}. Did you mean ${strings.dasherize(key)}?`);
+    }
+
+    camelCasedOptions[strings.camelize(key)] = value;
```

Comment:
Maybe split out the options here into `cliOptions and schematicOptions` and return both?  This could remove the need to loop through the options later on to remove the CLI-related options (those in `booleanArgs`).